docs/docs/migration-guide.mdx

```diff
@@ -24,6 +26,904 @@ If you still have training data in Markdown format then the recommended approach
+    ...
+  policies:
+    ...
+  ```
+  </TabItem>
+  <TabItem value="new">
+
+  ```yaml-rasa {1}
+  recipe: default.v1
+
+  language: en
+
+  pipeline:
+    ...
+  policies:
+    ...
+  ```
+  </TabItem>
+</Tabs>
+
+### Custom Policies and Custom Components
+
+Rasa Open Source 3.0 changed the way how [NLU components](components.mdx) and
+[policies](policies.mdx) are trained and run during inference. As part of these changes
+the interfaces for NLU components and policies have been unified and adapted.
+
+The next sections outline which adaptions are required to run your custom NLU components
+and policies with Rasa Open Source 3.0.
+
+:::caution
+
+Please read the updated guide on custom graph components [here](custom-graph-components.mdx) before
+continuing to follow the step-by-step guide to migrate your own custom graph components.
+:::
+
+**Type Annotations**
+
+Until Rasa Open Source 3.0 [type annotations](https://docs.python.org/3/library/typing.html)
+were not required in custom policies or custom NLU components.
+It is now required to use
+[type annotations](https://docs.python.org/3/library/typing.html) in custom NLU
+components and policies. Rasa Open Source uses these type annotations to validate that
+your graph components are compatible and correctly configured. As outlined in the custom
+[components guide](custom-graph-components.mdx) it is not allowed to use
+[Forward references](https://www.python.org/dev/peps/pep-0484/#forward-references).
+
+:::info Forward References with Python 3.7
+Use the
+[`from __future__ import annotations`](https://www.python.org/dev/peps/pep-0563/#enabling-the-future-behavior-in-python-3-7)
+import to avoid using forward references with Python 3.7.
+:::
+
+
+<Tabs values={[{"label": "Rasa 2.0 (old)", "value": "old"}, {"label": "Rasa 3.0 (new)", "value": "new"}]}  groupId="3-0-type-annotations" defaultValue="new">
+  <TabItem value="old">
+
+  ```python
+  class MyPolicy(Policy):
+      def train(
+          self,
+          training_trackers: List["TrackerWithCachedStates"],
+          domain,
+          **kwargs: Any,
+      ):
+          ...
+  ```
+
+  </TabItem>
+  <TabItem value="new">
+
+  ```python
+  from __future__ import annotations
+
+  class MyPolicy(Policy):
+      def train(
+          self,
+          training_trackers: List[TrackerWithCachedStates],
+          domain: Domain,
+          **kwargs: Any,
+      ) -> Resource:
+          ...
+  ```
+
+  </TabItem>
+</Tabs>
+
+#### Changes to Custom Policies
+This guide leads you through the migration of a custom policy step by step.
+
+**Instantiating a Policy for Training**
+
+Policies are no longer instantiated via their constructor. Instead, all policies have
+to implement a `create` method. During the policy instantiation the configuration from
+the [model configuration](model-configuration.mdx) is passed in as a dictionary instead
+of as separate parameters. Similarly, the`featurizers` are no longer instantiated
+outside of policies.
```

Comment:
I find this a bit confusing as in both examples a featurizer instance is passed in.

Reply:
better now? Can't really remove this param. Anything you would add to the existing `featurizer` description?